RustFxA/RustFirefoxAccounts.swift

```diff
@@ -50,15 +50,21 @@ open class RustFirefoxAccounts {
      hook into notifications like `.accountProfileUpdate` to refresh once initialize() is complete.
      Or they can wait on the accountManager deferred to fill.
      */
-    public static func startup(prefs _prefs: Prefs) -> Deferred<FxAccountManager> {
-        prefs = _prefs
-        if let manager = RustFirefoxAccounts.shared.accountManager.peek() {
-            return Deferred(value: manager)
+    public static func startup(prefs: Prefs) -> Deferred<FxAccountManager> {
+        assert(Thread.isMainThread)
+        RustFirefoxAccounts.prefs = prefs
+        if RustFirefoxAccounts.shared.accountManager.isFilled || isInitializingAccountManager {
+            return RustFirefoxAccounts.shared.accountManager
         }
 
-        let deferred = Deferred<FxAccountManager>()
+        // Setup the re-entrancy guard so consecutive calls to startup() won't do manager.initialize(). This flag is cleared when initialize is complete, or by calling reconfig().
+        isInitializingAccountManager = true
+
         let manager = RustFirefoxAccounts.shared.createAccountManager()
         manager.initialize { result in
+            assert(Thread.isMainThread)
```

Comment:
This is to help the reader, to see that non-thread-safe member vars are 100% safe to be used in this block. And it guards against any future bug in the rust lib that results in this completion being called off-main.